Core/HistoryManager.swift

```diff
@@ -17,21 +17,182 @@
 //  limitations under the License.
 //
 
+import CoreData
 import Foundation
 import BrowserServicesKit
+import History
+import Common
+import Persistence
 
-class HistoryManager {
+public protocol HistoryManaging {
 
-    let privacyConfig: PrivacyConfiguration
+    var historyCoordinator: HistoryCoordinating { get }
+
+}
+
+public class HistoryManager: HistoryManaging {
```

Comment:
Manages the history coordinator so we can swap it out if the privacy config changes in real time.